web-bundle/src/main/java/com/graphhopper/resources/RouteResource.java

```diff
@@ -112,6 +107,7 @@ public Response doGet(
                     + "or equal to the number of points (" + requestPoints.size() + ")");
 
         // TODO these checks should be only necessary once in the core, e.g. pointHints problems are currently ignored for POST requests
+        // todonow: maybe its time to fix this?
```

Comment:
Do you remember the reason this is done here instead of in `GraphHopper`? Can we move it over now?

Reply:
Good question. I do not remember a reason why this is here.

Reply:
Maybe for backward compatibility? Is it ok to add these checks for POST as well? If yes we can move them into `GraphHopper`.

Reply:
But there is more: Also the enableElevation, pointsEmpty checks and the "start heading" logic is missing for POST? All this should probably go into `GraphHopper#calcPaths` as well? Maybe this is also for another issue?

Reply:
Should be good for POST too, yes. Good idea with the new issue.